activerecord/test/cases/associations/has_many_associations_test.rb

```diff
@@ -3087,6 +3087,24 @@ def test_has_many_association_with_same_foreign_key_name
     end
   end
 
+  def test_key_ensuring_owner_was_is_not_valid_without_dependent_option
+    error = assert_raises(ArgumentError) do
+      Class.new(ActiveRecord::Base) do
+        has_many :books, ensuring_owner_was: true
```

Comment:
Passing `true` doesn't really fit with the use case for this, can we pass `:destroyed?` here? Yes, the code doesn't care, but it's more confusing to new readers.

```suggestion
        has_many :books, ensuring_owner_was: :destroyed?
```